pkg/skaffold/build/local/jib_maven.go

```diff
@@ -62,14 +68,46 @@ func (b *Builder) buildJibMavenToRegistry(ctx context.Context, out io.Writer, wo
 
 // generateMavenArgs generates the arguments to Maven for building the project as an image called `skaffoldImage`.
 func generateMavenArgs(goal string, skaffoldImage string, a *latest.JibMavenArtifact) []string {
-	command := []string{"prepare-package", "jib:" + goal, "-Dimage=" + skaffoldImage}
+	var command []string
+	if a.Module == "" {
+		// single-module project
+		command = []string{"--non-recursive", "prepare-package", "jib:" + goal, "-Dimage=" + skaffoldImage}
+	} else {
+		// multi-module project: we assume `package` is boujd to `jib:<goal>`
+		command = []string{"--projects", a.Module, "--also-make", "package", "-Dimage=" + skaffoldImage}
+	}
 	if a.Profile != "" {
-		command = append(command, "-P"+a.Profile)
+		command = append(command, "--activate-profiles", a.Profile)
 	}
 
 	return command
 }
 
+// verifyJibPackageGoal verifies that the referenced module has `package` bound to a single jib goal.
+// It returns `nil` if the goal is matched, and an error if there is a mismatch.
+func verifyJibPackageGoal(ctx context.Context, requiredGoal string, workspace string, a *latest.JibMavenArtifact) error {
+	// cannot use --non-recursive
+	command := []string{"--projects", a.Module, "jib:_skaffold-package-goals", "--quiet"}
+	if a.Profile != "" {
+		command = append(command, "--activate-profiles", a.Profile)
+	}
+
+	cmd := jib.MavenCommand.CreateCommand(ctx, workspace, command)
+	logrus.Debugf("Looking for jib bound package goals for %s: %s, %v", workspace, cmd.Path, cmd.Args)
+	stdout, err := util.RunCmdOut(cmd)
+	if err != nil {
+		return errors.Wrap(err, "could not obtain jib package goals")
+	}
+	goals := util.NonEmptyLines(stdout)
+	logrus.Debugf("jib bound package goals for %s %s: %v (%d)", workspace, a.Module, goals, len(goals))
+	if len(goals) != 1 {
+		return errors.New("skaffold requires a single jib goal bound to 'package'")
+	} else if goals[0] != requiredGoal {
+		return errors.New(fmt.Sprintf("skaffold `push` setting requires 'package' be bound to 'jib:%s'", requiredGoal))
```

Comment:
`fmt.Errorf("skaffold `push` setting requires 'package' be bound to 'jib:%s'", requiredGoal)` should work